mlrun/runtimes/utils.py

```diff
@@ -118,6 +119,31 @@ def resolve_spark_operator_version():
         raise ValueError("Failed to resolve spark operator's version")
 
 
+# if nuclio version specified on mlrun config set it likewise,
+# if not specified, get it from nuclio api client
+# since this is a heavy operation (sending requests to API), and it's unlikely that the version
+# will change - cache it (this means if we upgrade nuclio, we need to restart mlrun to re-fetch the new version)
+def resolve_nuclio_version():
+    from mlrun.api.utils.clients import nuclio
```

Comment:
What is the purpose of importing within the function?
We are mostly importing inside a function in two cases:
1. When a specific package is not required in the base requirements, but rather an additional feature that not all users will use.
2. To avoid circular imports.
The majority of imports within functions are in `config.py`.
Because config is the first to be initialized when we `import mlrun`, we couldn't import other modules in global scope until the config had been initialized, as most of our modules rely on it.